tsl/src/nodes/gapfill/planner.c

```diff
@@ -421,6 +426,16 @@ plan_add_gapfill(PlannerInfo *root, RelOptInfo *group_rel)
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("multiple time_bucket_gapfill calls not allowed")));
 
+	/* Executing time_bucket_gapfill on distributed hypertable produces incorrect result,
+	 * returns internal planning error or crashes if grouping includes a space dimension.
+	 * Thus time_bucket_gapfill is temporary disabled until it is fixed.
+	 */
+	if (dist_ht)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("support for time_bucket_gapfill on distributed hypertable is not "
+						"implemented")));
```

Comment:
I suggest simplifying to (with optional errdetail):

```suggestion
				 errmsg("gapfill not supported on distributed hypertable"),
				 errdetail("This limitation might be removed in the future."));
```